backend/app/models/project_involvement.rb

```diff
@@ -5,10 +5,14 @@ class ProjectInvolvement < ApplicationRecord
   validates_uniqueness_of :project_id, scope: :project_developer_id
 
   after_create do
-    ProjectDeveloperMailer.added_to_project(project_developer, project).deliver_later if project.published?
+    if project.valid? && project.published? && !project.status_changed?
```

Comment:
@martintomas  can you please add a comment to explain these conditions, i.e. the expected effect? just because I'm reading like 5th time and still not convinced I get it :D so for the `after_destroy`, we'll only notify them about being removed from project as collaborator if the destroy is not a result of a project destroy? And for the `after_create`, I don't even 😅 We only send the collaboration email if the project had already been published before a collaboration was added?

Reply:
Sure, I know it is complicated, didn't figure out better way to do it. For both callbacks applies that we send notification only when project is published and also valid. `after_destroy` is send only when `ProjectInvolvement` is not destroyed by Project itself (in such case we send notification to users that project was destroyed so no need to send extra mail that collaboration was removed). `after_create` is send only when status of project has not changed to `published` --> when you switch project status from `draft` to `published`, you send collaboration notification at once via Project callback so no need to send it twice. Basically tricky part here is that you can collaborators and project status via one update, but you need to send emails only once. Cha, will add comments :)